### DISCUSSION TITLE
some frequently asked questions

### DISCUSSION THREAD

**Discussion by kirakiray** [ofajs/discussion#3] [General]
- [ofa.js AI SKILL](https://ofajs.com/en/scenarios/ai-skill-usage.html)
- [Change attributes fails](https://github.com/kirakiray/ofa.js/issues/117)
- [How to know whether the default slot has child](https://github.com/kirakiray/ofa.js/issues/120)
- [Methods cannot be used within xfill](https://github.com/kirakiray/ofa.js/issues/115)

**Comment by prusswan** [comment]
This is a good idea, but you might want to include more ready-to-use examples (integration with some common UI component packages). I considered using this but managed to get Vue to work (no build step).

Also, is ofa something like orz? Easier for users to remember if there is a purpose/explanation behind the name

**Reply by kirakiray** [reply]
> This is a good idea, but you might want to include more ready-to-use examples (integration with some common UI component packages). I considered using this but managed to get Vue to work (no build step).
> 
> Also, is ofa something like orz? Easier for users to remember if there is a purpose/explanation behind the name

Thank you for your feedback, we are working hard to improve these points. Now we are seriously short of manpower, and it will be improved later. We are trying to use AI+ofajs to create some basic small tools, and then use the tools to quickly generate these document requirements. By then, the problems you mentioned can be improved, and the bugs of ofajs can be found and improved.
I will briefly answer your questions now:
* The UI component library is still under construction, the address is https://github.com/ofajs/Punch-UI, this library is still under reconstruction, and you can only refer to the demo when using it.
* Use ofa.js in vue, refer to https://github.com/ofajs/ofajs-in-vue3
* ofa means "one for all", which means you only need to write it once and it will be done once and for all.
